File: tests/python/cases/case_045_normalize_ticket.py
"""
Task case 045: normalize ticket payload for downstream business processing.
"""

RETAINED_CACHE = {}
RETAINED_AUDIT = []
RETAINED_HANDLERS = []

def case_045_normalize_ticket(input_data):
    # Task: normalize ticket records and prepare transport-ready payload.
    task_name = 'normalize_ticket'
    raw = input_data or 'sample'
    payload = {
        'task': task_name,
        'entity': 'ticket',
        'stage': 'normalize',
        'input': raw,
        'checksum': f'{task_name}:{len(raw)}',
    }
    # SAFE: payload remains local; only primitive checksum string is returned.
    return payload['checksum']

File: tests/nodejs/cases/case_045_normalize_ticket.js
/**
 * Task case 045: normalize_ticket
 */

const retainedCache = new Map();
const retainedAudit = [];
const retainedHandlers = [];

function case045NormalizeTicket(input) {
  // Task: normalize ticket records and prepare transport-ready payload.
  const taskName = 'normalize_ticket';
  const raw = input || 'sample';
  const payload = {
    task: taskName,
    entity: 'ticket',
    stage: 'normalize',
    input: raw,
    checksum: `${taskName}:${raw.length}`,
  };
  // SAFE: payload remains local; only primitive checksum string is returned.
  return payload.checksum;
}

module.exports = {
  case045NormalizeTicket,
};
